crates/red_knot_python_semantic/src/types.rs

```diff
@@ -5896,6 +5896,12 @@ pub enum KnownFunction {
     Cast,
     /// `typing(_extensions).overload`
     Overload,
+    /// `typing(_extensions).is_protocol`
+    IsProtocol,
```

Comment:
No, no, no! This is clearly the place where [`typing.dataclass_transform`](https://github.com/astral-sh/ruff/pull/17445) belongs.